### DISCUSSION TITLE
Prod squash migration asking to reset "public" schema

### DISCUSSION THREAD

**Discussion by SebastianGarces** [prisma/prisma#21835] [Q&A]
### Question

I'm trying to squash our prod migrations (eventually, using a testing env 1:1 to prod) but after creating the base migration, I'm being told that the database has applied migrations that are missing from the local migrations directory. 

I followed these instructions: [How to create a clean history in a production environment](https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/squashing-migrations#how-to-create-a-clean-history-in-a-production-environment)

Exact error:
```
The following migration(s) are applied to the database but missing from the local migrations directory: 
...(list of migrations), 0_base, 000000000000_base

We need to reset the "public" schema at "<db_location>"
Do you want to continue? All data will be lost
```

Already seen: 
https://github.com/prisma/docs/pull/3447

### Expected behavior (optional)

For squashed migration to actually be considered applied and new ones to come after.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @SebastianGarces 👋 

Thank you for raising this question.

This issue you are facing could be due to a discrepancy between the migration history in your database and the migration files in your local environment. To address this issue, you might need to use the `migrate resolve` command to mark the squashed migration as applied. Here's the command:
```
npx prisma migrate resolve --applied 000000000000_squashed_migrations
```
This command should mark the migration in the `000000000000_squashed_migrations` directory as already applied, preventing it from being run on production. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
